### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by YAXo-O**
+ ES6 Modules
+ Routing started (backwards/forwards in history vie browser buttons, changing URL)
+ MVC (Main menu, About Menu, Leaderboard = controllers + view, Base controllers and view; Views use template-based code, controllers wrap controlling widgets with Widget and Buttons classes, busines logic is implemented in controllers as well)
+ Herok link
+ MenuManager(router) and controllers use EventBus to 
a) break loop dependencies
b) to seperate different parts of architecture from each-other

### CODE REVIEW

File: Data/Game/CSS/GameStyle.css [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+.all_game {
+    position: absolute;
+    left: 20px;
+}
+
+.draggable {
+    cursor: pointer;
+}
+
+.table_field td
```
Comment (nksoff) [inline Data/Game/CSS/GameStyle.css:11]: не БЭМ

File: Data/Game/CSS/GameStyle.css [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+}
+
+.next_button
+{
+    position: absolute;
+    top: 65%;
+    left: 185%;
+}
+
+.turnButton
```
Comment (nksoff) [inline Data/Game/CSS/GameStyle.css:53]: где-то camelCase, где-то snake_case

File: Data/Modules/Controllers/LoginMenuController.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    }
+
+    submitHandler()
+    {
+        if(this.validate())
+            Services.checkUser(this.inputMail.value, this.inputPassword.value)
+                .then(response =>
+                {
+                    response;
+                    debugger;
```
Comment (nksoff) [inline Data/Modules/Controllers/LoginMenuController.js]: !

File: Data/Modules/Views/AboutMenuView/AboutMenuViewTemplate.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
+function pug_escape(e){var a=""+e,t=pug_match_html.exec(a);if(!t)return e;var r,c,n,s="";for(r=t.index,c=0;r<a.length;r++){switch(a.charCodeAt(r)){case 34:n="&quot;";break;case 38:n="&amp;";break;case 60:n="&lt;";break;case 62:n="&gt;";break;default:continue}c!==r&&(s+=a.substring(c,r)),c=r+1,s+=n}return c!==r?s+a.substring(c,r):s}
```
Comment (nksoff) [inline Data/Modules/Views/AboutMenuView/AboutMenuViewTemplate.js:1]: можно убрать это из репозитория?

Comment (YAXo-O) [reply Data/Modules/Views/AboutMenuView/AboutMenuViewTemplate.js:1]: Убрать все сгенеренные pug-ом шаблоны?

Comment (nksoff) [reply Data/Modules/Views/AboutMenuView/AboutMenuViewTemplate.js:1]: да

File: Data/Modules/Views/UserProfileBlockView/UserProfileBlock.css [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+    height: 80px;
+    border: 1px solid black;
+    position: fixed;
+    left: 10px;
+    top: 10px;
+    align-content: center;
+    text-align: center;
+}
+
+h3
```
Comment (nksoff) [inline Data/Modules/Views/UserProfileBlockView/UserProfileBlock.css:13]: ай-ай-ай

**Pull request merged**